web-client/public/js/update-app.js

```diff
@@ -48,6 +51,28 @@ export const updateApp = grnState => {
         grnState.newNetwork = false;
     }
 
+
+    if (grnState.normalizationMax !== null) {
+        var valueValidator = (min, max, value) => {
+            return Math.min(max, Math.max(min, value));
+        };
+
+        var edgeWeightNormalizationInputValidation = (value) => {
+            return value ===
+            "" ? "" : valueValidator(MIN_EDGE_WEIGHT_NORMALIZATION, MAX_EDGE_WEIGHT_NORMALIZATION, value);
+        };
+
+        var synchronizeNormalizationValues = (value) => {
+            var validated = edgeWeightNormalizationInputValidation(value);
+            $("#normalization-max").val(validated);
+            $("#edge-weight-normalization-factor-menu").val(validated);
+        };
+
+        synchronizeNormalizationValues(grnState.normalizationMax);
+        refreshApp();
```

Comment:
When those functions are moved out, then the part inside `updateApp` gets reduced to these two lines. Much more readable, yes?